cpp/common/src/codingstandards/cpp/Overflow.qll

```diff
@@ -0,0 +1,304 @@
+/**
+ * This module provides predicates for checking whether an operation overflows or wraps.
+ */
+
+import cpp
+import semmle.code.cpp.rangeanalysis.SimpleRangeAnalysis
+import SimpleRangeAnalysisCustomizations
+import semmle.code.cpp.controlflow.Guards
+import semmle.code.cpp.dataflow.TaintTracking
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+/**
+ * A `BinaryArithmeticOperation` which may overflow and is a potentially interesting case to review
+ * that is not covered by other queries for this rule.
+ */
+class InterestingBinaryOverflowingExpr extends BinaryArithmeticOperation {
```

Comment:
`BinaryArithmeticOperation` won't include any `AssignArithmeticOperation`, so variants of overflowing binary arithmetic operation operators that are assign arithmetic operation operators (`+=`, `-=`, `*=`, `/=` and `%=`) won't be included in the results. This same issue applies for some of the queries, e.g. INT33-C.

Reply:
Addressed